### DISCUSSION TITLE
rdkit.js: position of the atomic index is a bit strange, it may need to be  optimized

### DISCUSSION THREAD

**Discussion by ShangChien** [rdkit/rdkit#5195] [Development]
Version: rdkit.js_v2022.03.1
The display position of the atomic index is a bit strange, it may need to be  optimized.
The position of the index is a bit far from the position of the actual atom，as shown in the red circle below:
![index](https://user-images.githubusercontent.com/48350636/163103580-82e34db2-765e-468a-b6cd-f7249f406c03.PNG)

When I set to show both atoms and bonds,
```
addAtomIndices:true,
addBondIndices:true,
```
it can't be distinguish index of atoms and bonds accurately:

![all](https://user-images.githubusercontent.com/48350636/163104316-b84cadef-4a57-4de1-a8f2-67b1ad2297db.PNG)

The previous version did not have this problem (rdkit.js_v2021.09). 
I don't know the reason for this, but it's not a serious bug at the moment.

**Comment by DavidACosgrove** [comment]
Hi @ShangChien,
This is a consequence of adding indices to a molecule with highlights.  If you try it without highlights you will see that they are placed much more sensibly, as below.  When the indices are added, it tries to find somewhere where it doesn't clash with anything else that's on the drawing already.  If it can't find anywhere within its search space, it just gives up and puts it somewhere that isn't too bad.  It searches in a circle round the atom at increasing distances from it, and that's why you're sometimes seeing them some distance from the atom.  It is clearly not a great solution, particularly with the bond indices as well.  There are a few options:
1. Find a better fallback position.  This could be the first place it tries, which will at least be close in to the atom.
2. Do the search as at present, and if nowhere without a clash is found, try again ignoring the highlights.
3. Always ignore the highlights when doing the search.
In practice, 2 might end up looking like 3 in most cases.  All of these are possible without a great deal of effort, it's just a matter of deciding which one to go for.  @greglandrum, what do you think?

Cheers,
Dave
![image](https://user-images.githubusercontent.com/9198870/163180961-404f4a61-fdc6-4004-a765-649948fed2f4.png)
